pkg/kubectl/cmd/manifest/apply_manifest.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manifest
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+)
+
+type applyManifestOptions struct {
+	manifestDir string
+}
+
+type groupVersionKindName struct {
+	gvk schema.GroupVersionKind
+	// name of the resource.
+	name string
+}
+
+// NewCmdApplyManifest creates a new apply-manifest command.
+func NewCmdApplyManifest(f cmdutil.Factory, out, errOut io.Writer) *cobra.Command {
+	var options applyManifestOptions
+
+	cmd := &cobra.Command{
+		Use:   "apply-manifest [path]",
+		Short: "Use a Manifest file to apply a set of configurations",
+		Long:  templates.LongDesc("Use a Manifest file to apply a set of configurations"),
+		Example: templates.Examples(`
+		# Use the Kube-manifest.yaml file under somedir/ to apply a set of configurations.
+		kubectl alpha apply-manifest -f somedir/`),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(Validate(cmd, args))
+			cmdutil.CheckErr(Complete(cmd, args))
+			cmdutil.CheckErr(RunApplyManifest(f, cmd, out, errOut, &options))
+		},
+	}
+
+	cmd.Flags().StringVarP(&options.manifestDir, "filename", "f", "", "Pass in directory that contains the Kube-manifest.yaml file.")
+	cmd.MarkFlagRequired("filename")
+	cmd.Flags().StringP("output", "o", "yaml", "Output mode. Support json or yaml.")
+	return cmd
+}
+
+// Validate validates apply-manifest command.
+func Validate(cmd *cobra.Command, args []string) error {
+	return nil
+}
+
+// Complete completes apply-manifest command.
+func Complete(cmd *cobra.Command, args []string) error {
+	return nil
+}
+
+// RunApplyManifest runs apply-manifest command (do real work).
+func RunApplyManifest(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, options *applyManifestOptions) error {
```

Comment:
I would create an expand command that generates all the resources. The user could then decide whether to apply, create, get, delete, whatever.

Reply:
> I would create an expand command that generates all the resources. The user could then decide whether to apply, create, get, delete, whatever.

This is what this command does now.
I need to change the command name. Sorry for the confusion.